scripts/Tools/compare_test_results

```diff
@@ -87,8 +95,8 @@ def _main_func(description):
     baseline_name, baseline_root, test_root, compiler, test_id, compare_tests = \
         parse_command_line(sys.argv, description)
 
-    success = compare_test_results(baseline_name, baseline_root, test_root, compiler, test_id, compare_tests)
-    sys.exit(0 if success else 1)
+    compare_test_results(baseline_name, baseline_root, test_root, compiler, test_id, compare_tests)
```

Comment:
I would like to preserve the behavior of a non-zero error code for diffs. One thing I've done in many places in CIME (which I probably should have done here too) is to return CIME.utils.TESTS_FAILED_ERR_CODE when everything works but there are diffs. See create_test as an example.